From 708f25cfe522df162c3e2c7c56cbe2f7000cb5e4 Mon Sep 17 00:00:00 2001
From: Gary Bisson <gary.bisson@boundarydevices.com>
Date: Mon, 14 Dec 2020 14:26:01 +0100
Subject: [PATCH] start_isp.sh: fix test to be generic

So that it works on EVK and any other i.MX 8M Plus platform.

Signed-off-by: Gary Bisson <gary.bisson@boundarydevices.com>
---
 imx/start_isp.sh | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

Index: isp-imx-4.2.2.11.0/imx/start_isp.sh
===================================================================
--- isp-imx-4.2.2.11.0.orig/imx/start_isp.sh
+++ isp-imx-4.2.2.11.0/imx/start_isp.sh
@@ -8,6 +8,7 @@
 
 RUNTIME_DIR="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
 NR_DEVICE_TREE_BASLER=$(grep basler-camera-vvcam /sys/firmware/devicetree/base/soc@0/*/i2c@*/*/compatible -l | wc -l 2> /dev/null)
+DMESG_BASLER=`dmesg | grep Basler`
 
 # check if the basler device has been enabled in the device tree
 if [ $NR_DEVICE_TREE_BASLER -eq 1 ]; then
@@ -28,8 +29,15 @@ elif [ $NR_DEVICE_TREE_BASLER -eq 2 ]; t
 	# Available configurations: dual_basler_1080p60, dual_basler_1080p60hdr
 	exec ./run.sh -c dual_basler_1080p60 -lm
 
+elif [ ! -z "$DMESG_BASLER" ]; then
+
+	echo "Starting isp_media_server for generic Basler"
+
+	cd $RUNTIME_DIR
+	exec ./run.sh -c basler_1080p60
+
 else
 	# no device tree found exit with code no device or address
-	echo "No device tree found for Basler, check dtb file!" >&2
+	echo "Basler camera not found!" >&2
 	exit 6
 fi
